src/components/Tooltip/story.js

```diff
@@ -0,0 +1,251 @@
+/* eslint-disable import/no-anonymous-default-export */
+import React, { Fragment, useState } from 'react';
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import {
+  boolean,
+  number,
+  optionsKnob as options,
+  text,
+} from '@storybook/addon-knobs';
+
+import Tooltip, { EasyRichTooltip, directions } from './';
+import Button from '../Button';
+import Readme from './README.md';
+
+const TooltipContent = (
+  <Fragment>
+    <h3 style={{ margin: 0 }}>Hallo there!</h3>
+    <p>
+      Pellentesque habitant morbi tristique senectus et netus et malesuada fames
+      ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget,
+      tempor sit amet, ante. Donec eu libero sit amet quam egestas semper.
+      Aenean
+    </p>
+  </Fragment>
+);
+
+const RichTooltip = ({ children, ...rest }) => {
+  // eslint-disable-next-line react/prop-types
+  const render = ({ toggle }) => (
+    <Button onClick={toggle} style={{ marginRight: 10 }}>
+      {children}
+    </Button>
+  );
+
+  return <EasyRichTooltip {...rest}>{render}</EasyRichTooltip>;
+};
+
+RichTooltip.propTypes = {
+  ...Tooltip.propTypes,
+};
+
+const FlexWrapper = props => (
+  <div
+    style={{
+      display: 'flex',
+      justifyContent: 'center',
+      alignContent: 'center',
+      alignItems: 'center',
+      padding: '20px',
+      height: '50vh',
+    }}
+    {...props}
+  />
+);
+
+const Element = props => <div style={{ marginRight: 20 }} {...props} />;
+
+const Overview = () => {
+  const [open, setOpen] = useState(false);
+
+  const toggle = () => {
+    setOpen(prev => !prev);
+  };
+
+  const direction = options(
+    'Direction',
+    directions,
+    directions[0],
+    {
+      display: 'inline-radio',
+    },
+    'all'
+  );
+  const textHover = boolean('Show on hover', true, 'text');
+
+  return (
+    <FlexWrapper>
+      <Element>
+        <Tooltip
+          content={text('Tooltip Text', 'Hi there!', 'text')}
+          showOnHover={textHover}
+          direction={direction}
+          isOpen={boolean('Open', false, 'text')}
+        >
+          <Button>{text('Main Content', 'With Text', 'text')}</Button>
+        </Tooltip>
+      </Element>
+      <Tooltip
+        isOpen={open}
+        showOnHover={boolean('Show on hover', false, 'content')}
+        direction={direction}
+        closeable={toggle}
```

Comment:
what do you guys think about having an `onClose` function instead of using this as a toggling function? I feel like by giving this to the user we give them space to use them in the wrong way. Let's say, for example, the user starts with `isOpen` as `true` and then ` <Button onClick={toggle}>With content</Button>` gets clicked, the tooltip will close instead of open, it might sound like a contrived example, and that a user will not do that, but by exposing this API we let them do it.

I'm not proposing to remove this prop, just converting it to a boolean and adding an `onClose` prop.

The EasyRichTooltip would look like this:

```typescript
export const EasyRichTooltip = ({ closeable, onClose = () => {}, children, isOpen = false, ...props }) => {
  const [isOpen, setOpen] = useState(isOpen);

  const open = () => setOpen(true);
  const close = () => { 
    setOpen(false);
    onClose();
  };
  const toggle = () => {
    if (isOpen) {
      close();
    } else. {
      open();
    }
  };

  return (
    <Tooltip
      {...props}
      closeable={closeable}
      onClose={onClose}
      isOpen={isOpen} // I moved it after the spread to avoid overwriting it
    >
      {children({ open, close, toggle })}
    </Tooltip>
  );
};
```

If we decide not to go with this approach I think we should proxy the `isOpen` proxy and also move the props' spread to the top.

cc: @mdespuits @pixelbandito

Reply:
I've since changed this to `onClose`. I figured since `closeable` will _always_ require an `onClose` behavior (practically speaking) it made sense to use the existence of a function passed into `onClose` as enough to enable that behavior.

I'm not absolutely married to it, but it does reduce the property surface area of the tooltip a bit.